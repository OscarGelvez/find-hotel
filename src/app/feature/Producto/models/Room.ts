export interface IRoom {
  /**
   * Identificador del hotel
   */
  id: number;
  /**
   * Objeto hotel al cual pertenece la habitación
   */
  hotel: IHotel;
  /**
   * Valor de la habitación
   */
  value: number;
  /**
   * Indica si tiene o no aire acondicionado
   */
  ac: string;
  /**
   * Indica si cuenta o no con parquedadero
   */
  parking: string;
  /**
   * Fecha inicial de disponibilidad
   */
  available_from: string;
  /**
   * Fecha final de disponibilidad
   */
  available_until: string;
  /**
   * Indica si cuenta con conexión wifi
   */
  wifi: string;
  /**
   * Capacidad de personas por habitación
   */
  capacity: number;
  /**
   * Descripción general de la habitación
   */
  description: string;
}

export interface IHotel {
  /**
   * Identificador del hotel
   */
  id: number;
  /**
   * Nombre del hotel
   */
  name: string;
  /**
   * Dirección y/o ubicacion del hotel
   */
  address: string;
  /**
   * Cantidad de estrellas del hotel
   */
  stars: number;
  /**
   * Descripción general
   */
  description: string;
  /**
   * Puntaje de calificación dado por usuarios
   */
  score: number;
}
