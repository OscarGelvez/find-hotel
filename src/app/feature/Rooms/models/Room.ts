export interface IRoom {
  /**
   * Identificador del hotel
   */
  id: number;
  /**
   * Título de la habitación
   */
  title: string;
  /**
   * Imagen descriptiva de la habitación
   */
  image: string;
  /**
   * Objeto hotel al cual pertenece la habitación
   */
  hotel: IHotel;
  /**
   * Valor de la habitación
   */
  value: string;
  /**
   * Indica si tiene o no aire acondicionado
   */
  ac: string;
  /**
   * Indica si cuenta o no con parquedadero
   */
  parking: string;
  /**
   * Fecha inicial de disponibilidad
   */
  available_from: string;
  /**
   * Fecha final de disponibilidad
   */
  available_until: string;
  /**
   * Indica si cuenta con conexión wifi
   */
  wifi: string;
  /**
   * Capacidad de personas por habitación
   */
  capacity: number;
  /**
   * Descripción general de la habitación
   */
  description: string;
}

export interface IHotel {
  /**
   * Identificador del hotel
   */
  id: number;
  /**
   * Nombre del hotel
   */
  name: string;
  /**
   * Dirección y/o ubicacion del hotel
   */
  address: string;
  /**
   * Cantidad de estrellas del hotel
   */
  stars: number;
  /**
   * Descripción general
   */
  description: string;
  /**
   * Puntaje de calificación dado por usuarios
   */
  score: number;
}

export interface IFieldsForm {
  /**
   * Fecha inicio de reservación
   */
  fromDate?: string;
  /**
   * Fecha final de reservación
   */
  untilDate?: string;
  /**
   * Cantidad de personas en habitación
   */
  capacity?: string;
  /**
   * Habitación con aire acondicionado
   */
  withAir?: boolean;
  /**
   * Habitación con parqueadero propio
   */
  withParking?: boolean;
}

export interface IFieldsFormBooking {
  /**
   * Nombre de ususario que hace la reserva
   */
  name: string;
  /**
   * Correo electrónico de quien hacer la reserva
   */
  email: string;
  /**
   * Tipo de identificación
   */
  identification_type: string;
  /**
   * Número de identificación
   */
  identification: string;
}

export interface IBookingData {
  /**
   * Objeto con información de la habitación seleccionada
   */
  roomData: IRoom;
  /**
   * Objeto con la información el usuari oque reserva
   */
  bookData: IFieldsFormBooking;
}

export interface IFieldsFormFindBooking {
  /**
   * Correo electrónico de quien hacer la reserva
   */
  email: string;
  /**
   * Tipo de identificación
   */
  identification_type: string;
  /**
   * Número de identificación
   */
  identification: string;
}
