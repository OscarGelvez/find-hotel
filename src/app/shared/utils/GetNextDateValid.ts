/**
 * Valida la siguiente fecha valida del campo Fecha de salida
 * según la fecha seleccionada en el campo Fecha ingreso
 * @param currentDate Fecha actual
 * @param fromDate Fecha inicial
 * @param minDate mínima fecha incial
 * @returns
 */
export const getNextDate = (
  currentDate: Date,
  fromDate: string,
  minDate: string
) => {
  if (fromDate === minDate) {
    currentDate.setDate(currentDate.getDate() + 1);
    return currentDate.toISOString().split('T')[0];
  } else {
    const auxDate = fromDate !== '' ? new Date(fromDate) : currentDate;
    auxDate.setDate(auxDate.getDate() + 1);
    return auxDate.toISOString().split('T')[0];
  }
};
