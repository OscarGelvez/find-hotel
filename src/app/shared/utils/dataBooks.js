export const bookInfo = {
  bookData: {
    name: 'oscar andres',
    email: 'oscargelvez23@gmail.com',
    identification_type: '1',
    identification: '1323131232',
  },
  roomData: {
    id: 54,
    title: 'Peace and Love',
    image:
      'https://dynamic-media-cdn.tripadvisor.com/media/photo-o/0d/2d/70/6d/habitacion-master-suite.jpg?w=600&h=300&s=1',
    hotel: {
      id: 3,
      name: 'Cencor hotel',
      address: 'Calle 1 # 16-23',
      stars: 4,
      description: 'Ambiente familiar, amigos, descaso y relajaci칩n',
      score: 7.2,
    },
    value: '56.000',
    ac: 'N',
    parking: 'N',
    available_from: '2021-05-02',
    available_until: '2021-05-04',
    wifi: 'N',
    capacity: 1,
    description: 'Ba침o privado, cama sencilla, TV.',
  },
  id: 5,
};

export const bookInfo2 = {
  bookData: {
    name: 'Oscar Gelvez Test',
    email: 'test@cypress.com',
    identification_type: '1',
    identification: '123456789',
  },
  roomData: {
    id: 50,
    title: 'Suite full. Caroline',
    image: 'https://static.abc.es/Media/201504/27/hotel12--644x362.jpg',
    hotel: {
      id: 1,
      name: 'Gold Hotel 5',
      address: 'Calle 54 # 56 -32 Av Libre',
      stars: 4,
      description:
        'Hotel con ambiente familiar, increible opci칩n. Muchas comodidades',
      score: 8.7,
    },
    value: '154.000',
    ac: 'Y',
    parking: 'Y',
    available_from: '2021-04-13',
    available_until: '2021-04-30',
    wifi: 'Y',
    capacity: 1,
    description: 'Ba침o privado, cama doble, con nevera, Wifi, TV.',
  },
  id: 1,
};
