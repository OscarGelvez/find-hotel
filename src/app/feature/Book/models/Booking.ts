import { IRoom } from 'app/feature/Rooms/models/Room';

export interface IFieldsFormBooking {
  /**
   * Nombre de ususario que hace la reserva
   */
  name: string;
  /**
   * Correo electrónico de quien hacer la reserva
   */
  email: string;
  /**
   * Tipo de identificación
   */
  identification_type: string;
  /**
   * Número de identificación
   */
  identification: string;
  /**
   * Fecha de ingreso al hotel
   */
  fromDateBook: string;
  /**
   * Fecha de salida del hotel
   */
  untilDateBook: string;
}

export interface IBooking {
  /**
   * Objeto con información de la habitación seleccionada
   */
  roomData: IRoom;
  /**
   * Objeto con la información el usuari oque reserva
   */
  bookData: IFieldsFormBooking;
  /**
   * Identificador en base de datos
   */
  id?: number;
}

export interface IFieldsFormFindBooking {
  /**
   * Correo electrónico de quien hacer la reserva
   */
  email: string;
  /**
   * Tipo de identificación
   */
  identification_type: string;
  /**
   * Número de identificación
   */
  identification: string;
}
