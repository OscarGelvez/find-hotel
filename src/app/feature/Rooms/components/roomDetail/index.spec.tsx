import '@testing-library/jest-dom';

import { cleanup, render } from '@testing-library/react';
import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import { roomInfo } from '../../../../shared/utils/data';
import { IRoom } from '../../models/Room';
import { RoomDetailView } from './index';

const newRoom: IRoom = roomInfo;

const roomId = 50;
afterEach(cleanup);

it('rederizar el componente RoomDetailView sin errores', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <BrowserRouter>
      <RoomDetailView
        listRooms={[newRoom]}
        selectedRoom={roomId}
        saveBookingRoom={() => null}
        isLoading={false}
      />
    </BrowserRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('renderiza el componente RoomDetailView y verifica existencia de componentes hijos', () => {
  const { getByTestId } = render(
    <BrowserRouter>
      <RoomDetailView
        listRooms={[newRoom]}
        selectedRoom={roomId}
        saveBookingRoom={() => null}
        isLoading={false}
      />
    </BrowserRouter>
  );
  expect(getByTestId('room-detail-view')).toContainElement(
    getByTestId('card-information')
  );
});

it('Compara snapshot del componente RoomDetailView', () => {
  const element = renderer
    .create(
      <BrowserRouter>
        <RoomDetailView
          listRooms={[newRoom]}
          selectedRoom={roomId}
          saveBookingRoom={() => null}
          isLoading={false}
        />
      </BrowserRouter>
    )
    .toJSON();
  expect(element).toMatchSnapshot();
});
