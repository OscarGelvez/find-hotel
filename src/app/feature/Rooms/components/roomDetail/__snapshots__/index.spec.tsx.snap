// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compara snapshot del componente RoomDetailView 1`] = `
<div
  className="card card-detail-screen"
>
  <div
    className="mt-3 ml-3 text-center"
  >
    <h3>
      Detalle de habitación
    </h3>
  </div>
  <div
    className="card-rooms"
    data-testid="room-detail-view"
  >
    <div
      className="row pt-5"
    >
      <div
        className="col"
      >
        <div
          className="card room-item"
          data-testid="card-information"
        >
          <div
            className="card-header room-item-header "
          >
            <h3>
              Gold Hotel 5
            </h3>
            <h4>
              8.7
            </h4>
          </div>
          <div
            className="card-body"
          >
            <div
              className="row"
            >
              <div
                className="col col-md-4"
              >
                <img
                  alt="hotel room"
                  className="img-fluid"
                  height="180px"
                  src="https://static.abc.es/Media/201504/27/hotel12--644x362.jpg"
                  width="260px"
                />
              </div>
              <div
                className="col col-md-8"
              >
                <div
                  className="row"
                >
                  <div
                    className="col-12 col-lg-8"
                  >
                    <h5
                      className="card-title"
                    >
                      Suite full. Caroline
                    </h5>
                  </div>
                  <div
                    className="col-12 col-lg-4"
                  >
                    <button
                      className="btn btn-primary btn-book w-100"
                      data-bs-target="#modalFormBook"
                      data-bs-toggle="modal"
                      data-testid="form-detail-button-book"
                      id="form-detail-button"
                      type="button"
                    >
                      <i
                        className="far fa-calendar-check ml-3"
                      />
                       Reservar
                    </button>
                    <div
                      data-testid="modal-form-book"
                    >
                      <div
                        aria-labelledby="modalFormBookLabel"
                        className="modal fade"
                        data-testid="modal-book"
                        id="modalFormBook"
                      >
                        <div
                          className="modal-dialog"
                        >
                          <div
                            className="modal-content"
                          >
                            <div
                              className="modal-header"
                            >
                              <h5
                                className="modal-title"
                                id="modalFormBookLabel"
                              >
                                Completa tu reserva
                              </h5>
                              <button
                                aria-label="Close"
                                className="btn-close"
                                data-bs-dismiss="modal"
                                type="button"
                              />
                            </div>
                            <div
                              className="modal-body"
                            >
                              <form
                                action="#"
                                data-testid="modal-book-form"
                                id="form-modal-book"
                                onReset={[Function]}
                                onSubmit={[Function]}
                              >
                                <div
                                  className="row mb-3"
                                >
                                  <div
                                    className="col-12 col-md-6 "
                                  >
                                    <label
                                      className="col-12 col-form-label"
                                      htmlFor="name"
                                    >
                                      Nombre
                                    </label>
                                    <input
                                      className="form-control col-12"
                                      data-testid="modal-book-form-name"
                                      id="book-form-name"
                                      name="name"
                                      onBlur={[Function]}
                                      onChange={[Function]}
                                      placeholder="eje: Joe Doe"
                                      value=""
                                    />
                                  </div>
                                  <div
                                    className=" col-12 col-md-6"
                                  >
                                    <label
                                      className="col-12 col-form-label"
                                      htmlFor="email"
                                    >
                                      Correo electrónico
                                    </label>
                                    <input
                                      className="form-control col-12"
                                      data-testid="modal-book-form-email"
                                      id="book-form-email"
                                      name="email"
                                      onBlur={[Function]}
                                      onChange={[Function]}
                                      placeholder="eje: joe@email.com"
                                      type="email"
                                      value=""
                                    />
                                  </div>
                                </div>
                                <div
                                  className="row mb-2"
                                >
                                  <div
                                    className="col-12 col-md-6 "
                                  >
                                    <label
                                      className="pr-2"
                                      htmlFor="identification_type"
                                    >
                                      Tipo de identificación
                                    </label>
                                    <select
                                      aria-label=".form-select-sm example"
                                      className=" d-block w-100"
                                      data-testid="modal-book-form-identification-type"
                                      id="book-form-select"
                                      name="identification_type"
                                      onBlur={[Function]}
                                      onChange={[Function]}
                                      value=""
                                    >
                                      <option
                                        value=""
                                      >
                                        Seleccione
                                      </option>
                                      <option
                                        value="1"
                                      >
                                        Cédula de ciudadanía
                                      </option>
                                      <option
                                        value="2"
                                      >
                                        Cédula de extranjeria
                                      </option>
                                      <option
                                        value="3"
                                      >
                                        Nit
                                      </option>
                                      <option
                                        value="4"
                                      >
                                        Pasaporte
                                      </option>
                                    </select>
                                  </div>
                                  <div
                                    className=" col-12 col-md-6"
                                  >
                                    <label
                                      className="pr-2"
                                      htmlFor="identification"
                                    >
                                      Identificación
                                    </label>
                                    <input
                                      className="form-control"
                                      data-testid="modal-book-form-identification"
                                      id="book-form-identification"
                                      name="identification"
                                      onBlur={[Function]}
                                      onChange={[Function]}
                                      placeholder="eje: 60830147"
                                      value=""
                                    />
                                  </div>
                                </div>
                                <div
                                  className="d-flex justify-content-between mt-3"
                                >
                                  <button
                                    className="btn btn-secondary"
                                    data-bs-dismiss="modal"
                                    data-testid="modal-book-form-btn-cancel"
                                    id="modal-form-btn-cancel"
                                    type="button"
                                  >
                                    Cancelar
                                  </button>
                                  <button
                                    className="btn btn-primary"
                                    data-testid="modal-book-form-btn-submit"
                                    id="modal-form-btn-submit"
                                    type="submit"
                                  >
                                    Finalizar
                                  </button>
                                </div>
                              </form>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    className="col-12 col-md-12"
                  >
                    <h4>
                      <i
                        className="far fa-money-bill-alt"
                      />
                       
                      154.000
                    </h4>
                  </div>
                  <div
                    className="col-12 col-md-12 d-flex"
                  >
                    <h6
                      className="font-weight-bold"
                      data-testid="custom-field"
                    >
                      Inicio
                       
                      <i
                        className="fas fa-calendar-alt"
                      />
                       :
                       
                      <span
                        data-testid="custom-field-value-text"
                      >
                        2021-04-13
                      </span>
                    </h6>
                    <h6
                      className="font-weight-bold"
                      data-testid="custom-field"
                    >
                      -  Fin
                       
                      <i
                        className="fas fa-calendar-alt"
                      />
                       :
                       
                      <span
                        data-testid="custom-field-value-text"
                      >
                        2021-04-30
                      </span>
                    </h6>
                  </div>
                </div>
                <p
                  className="card-text"
                >
                  Baño privado, cama doble, con nevera, Wifi, TV.
                </p>
                <div
                  className="mb-3"
                  data-testid="form-field-hotel-desc"
                >
                  <p
                    className="card-text"
                  >
                    Hotel con ambiente familiar, increible opción. Muchas comodidades
                  </p>
                  <h6
                    className="font-weight-bold"
                    data-testid="custom-field"
                  >
                    Dirección
                     
                    <i
                      className="fas fa-map-marked-alt"
                    />
                     :
                     
                    <span
                      data-testid="custom-field-value-text"
                    >
                      Calle 54 # 56 -32 Av Libre
                    </span>
                  </h6>
                </div>
                <div
                  className="row"
                >
                  <div
                    className="col col-md-6"
                  >
                    <h6
                      className="font-weight-bold"
                      data-testid="custom-field"
                    >
                      Camas
                       
                      <i
                        className="fas fa-bed"
                      />
                       :
                       
                      <span
                        data-testid="custom-field-value-text"
                      >
                        1
                      </span>
                    </h6>
                    <h6
                      className="font-weight-bold"
                      data-testid="custom-field"
                    >
                      Aire
                       
                      <i
                        className="fas fa-wind"
                      />
                       :
                       
                      <i
                        className="fas fa-check"
                        data-testid="custom-field-value-true"
                      />
                    </h6>
                    <h6
                      className="font-weight-bold"
                      data-testid="custom-field"
                    >
                      Wifi
                       
                      <i
                        className="fas fa-wifi"
                      />
                       :
                       
                      <i
                        className="fas fa-check"
                        data-testid="custom-field-value-true"
                      />
                    </h6>
                  </div>
                  <div
                    className="col col-md-6"
                  >
                    <h6
                      className="font-weight-bold"
                      data-testid="custom-field"
                    >
                      Parqueadero
                       
                      <i
                        className="fas fa-car"
                      />
                       :
                       
                      <i
                        className="fas fa-check"
                        data-testid="custom-field-value-true"
                      />
                    </h6>
                    <h6
                      className="font-weight-bold"
                      data-testid="custom-field"
                    >
                      Baños
                       
                      <i
                        className="fas fa-restroom"
                      />
                       :
                       
                      <i
                        className="fas fa-check"
                        data-testid="custom-field-value-true"
                      />
                    </h6>
                    <div
                      data-testid="star-score"
                    >
                      <i
                        className="fas fa-star"
                      />
                      <i
                        className="fas fa-star"
                      />
                      <i
                        className="fas fa-star"
                      />
                      <i
                        className="fas fa-star"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="w-50 d-block p-4"
  >
    <a
      className="sc-AxjAm jqjyPd"
      href="/rooms"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary w-25 ml-4"
        data-testid="btn-back-detail"
        id="btn-back-detail"
        onClick={[Function]}
        type="button"
      >
        <i
          className="fas fa-arrow-left"
        />
         Atras
      </button>
    </a>
  </div>
</div>
`;
